src/transformers/models/gemma2/modeling_gemma2.py

```diff
@@ -818,19 +811,8 @@ def forward(
         if inputs_embeds is None:
             inputs_embeds = self.embed_tokens(input_ids)
 
-        if cache_position is None:
-            if past_key_values is None:
-                cache_position = torch.arange(0, inputs_embeds.shape[1], device=inputs_embeds.device)
-            else:
-                raise ValueError("When `past_key_values` is passed, `cache_position` must be too")
-
-        # Probably a forward call with caching, so we set up cache for one call only
-        if use_cache and past_key_values is None and not self.training:
```

Comment:
Two changes here, both to be consistent with other models:
1. `self.training` should not control whether we instantiate a cache
2. If a user respects the types in the docs, `past_key_values` is either a `Cache` or we instantiate a new one for the user without warnings